hosts/galileo/ssi/__init__.py

```diff
@@ -0,0 +1,334 @@
+
+    # Define the field of view
+    FOV = meta.fov(full_fov=full_fov)
+
+    # Trim the image
+    data = meta.trim(vic.data_2d, full_fov=full_fov)
+
+    # Create a Snapshot
+    result = oops.obs.Snapshot(('v','u'), meta.tstart, meta.exposure,
+                               FOV,
+                               path = 'GLL',
+                               frame = 'GLL_SCAN_PLATFORM',
+                               dict = vicar_dict,       # Add the VICAR dict
+                               data = data,             # Add the data array
+                               instrument = 'SSI',
+                               filter = meta.filter,
+                               filespec = filespec,
+                               basename = os.path.basename(filespec))
+
+    result.insert_subfield('spice_kernels',
+                           Galileo.used_kernels(result.time, 'iss',
+                                                return_all_planets))
+
+    return result
+
+#===============================================================================
+def initialize(ck='reconstructed', planets=None, asof=None,
+               spk='reconstructed', gapfill=True,
+               mst_pck=True, irregulars=True):
+    """Initialize key information about the SSI instrument.
+
+    Must be called first. After the first call, later calls to this function
+    are ignored.
+
+    Input:
+        ck,spk      'predicted', 'reconstructed', or 'none', depending on which
+                    kernels are to be used. Defaults are 'reconstructed'. Use
+                    'none' if the kernels are to be managed manually.
+        planets     A list of planets to pass to define_solar_system. None or
+                    0 means all.
+        asof        Only use SPICE kernels that existed before this date; None
+                    to ignore.
+        gapfill     True to include gapfill CKs. False otherwise.
+        mst_pck     True to include MST PCKs, which update the rotation models
+                    for some of the small moons.
+        irregulars  True to include the irregular satellites;
+                    False otherwise.
+    """
+    SSI.initialize(ck=ck, planets=planets, asof=asof,
+                   spk=spk, gapfill=gapfill,
+                   mst_pck=mst_pck, irregulars=irregulars)
+
+
+#===============================================================================
+class Metadata(object):
+
+    #===========================================================================
+    def __init__(self, label):
+        """Use the label to assemble the image metadata.
+
+        Input:
+            label           The label dictionary.
+
+        Attributes:
+            nlines          A Numpy array containing the data in axis order
+                            (line, sample).
+            nsamples        The time sampling array in (line, sample) axis
+                            order, or None if no time backplane is found in
```

Comment:
Does Galileo have a time backplane?

Reply:
This comment is not even relevant; removed